### PR TITLE
Map improvements and general fixes

### PR DESCRIPTION
**Pull request by MichaelEbert**
Nirnroot map improvements

    Made CreateLinkElement use the same CELL_FORMAT modifiers as initSingleCell
    Added new CELL_FORMAT modifiers specific to CreateLinkElement
    Nirnroot will now link to the map instead of to nothing
    If map is on nirnroot, it will load the nirnroots instead of always loading locations first
    Nirnroots can be selected
    If map target is a nirnroot, map will show nirnroots, not locations.

moved saves to last, so they show up under everything else.

changed spells to dark orange to show up better. Added link to map on guide page, fixing #135

map formatting fixes. Added 'map' title to map page.

added auto text settings. Added iframe min width settings, fixing #87 to my satisfaction. Added iframe saved width, fixing #101. Set default iframe width to 45vw, so it scales with window size.

changed topbar to sticky, so it'll be above the map. Also, removing the topbar will now make the map use up that space instead of just leaving it blank.

Changed buttons to radio buttons on map page

split out gates into a separate class so we can draw them differently. Updated gate overlay icons to be more easily readable

### CODE REVIEW

File: js/map/mapObject.mjs [hunk 1]
```diff
@@ -202,8 +194,45 @@ MapIcon.prototype.onClick = function(clickPos){
...
+    MapIcon.call(this, cell);
+    this.isRandom = cell.notes?.includes("Random") ?? false;
+}
+
+GateIcon.prototype = Object.create(MapIcon.prototype);
+
+GateIcon.prototype.draw = function(ctx, mouseLoc, currentSelection){
+    if(this.isRandom){
+        if(getRandomGateCount() >= 40){
+            //draw in grey? idk.
```
Comment (Muggyfox1) [inline js/map/mapObject.mjs:211]: we could invert the gate color like so:
Icon_Gate inverted

Or put in another slightly transparent gate icon like so:
Icon_Gate_Ghost

I remember these being mentioned as ideas to solve the problem before.

File: map.html [hunk 2]
```diff
@@ -69,11 +69,15 @@
     <div class="topbarSection"><span class="totalProgressPercent">0</span>%</div> 
 </div>
 <div class="mapContainer" id="mapContainer">
+	<h1 id="title">Map</h1>
```
Comment (Muggyfox1) [inline map.html:72]: Do we still need the title map on here if the majority of the page is the map? 
it seemed self-evident enough(to me) to not have the title.

**Review by Muggyfox1 — COMMENTED** [review]
Lots of good work. Good job!

File: js/map/mapObject.mjs (hunk 1 above)
Comment (Muggyfox1) [reply js/map/mapObject.mjs:211]: oh and also, would this be related to and resolve #136 ?

Comment (MichaelEbert) [reply js/map/mapObject.mjs:211]: Yeah. If you want to work on that, that's a good thing to do.

File: map.html (hunk 2 above)
Comment (MichaelEbert) [reply map.html:72]: All the other pages have it, so I figured why not. But you are right that it's not really necessary.

**Pull request merged**